sdcm/commit_log_check_thread.py

```diff
@@ -0,0 +1,190 @@
+        this condition will ignore result when Scylla does not exceed  segments limit
+
+        - {commitlog_segment_size_in_mb}*1048576) :
+        minus size of 1 segment,  to ensure that Scylla has at least 1 free segment number of active_bytes should be
+        less then "scylla_commitlog_disk_total_bytes"-"1 segment"
+
+        scylla_commitlog_disk_active_bytes > :
+        filter all good values when scylla_commitlog_disk_active_bytes<= scylla_commitlog_disk_total_bytes - 1 segment
+        and return from Prometheus only bad
+    """
+
+
+# pylint: disable=too-many-instance-attributes
+class CommitLogCheckThread:
+    """
+        if commitlog-use-hard-size-limit is enabled,
+        this thread will check the following metrics during the test:
+
+        1) commit log directory does not exceed the limit at any stage of the test.
+           if it exceeded - send an error event that fails the test.
+
+        2) free segments don't drop to zero at any stage of the test,
+        otherwise, send an error event that fails the test.
+    """
+
+    # Prometheus API limit: check_interval/discreteness must be less then 11000
+    check_interval = 10 * 60
+    discreteness = 1
+
+    def __init__(self, custer_tester, test_duration, termination_event=None, thread_name: str = ""):
+        self.log = logging.getLogger(self.__class__.__name__)
+        self.prometheus = custer_tester.prometheus_db
+        self.test_duration = test_duration
+
+        self.commitlog_params = CommitlogConfigParams(custer_tester.db_cluster)
+        self.prometheus_queries = PrometheusQueries(self.commitlog_params)
+        self.start_time = None
+
+        self._thread = threading.Thread(daemon=True, name=f"{self.__class__.__name__}_{thread_name}", target=self.run)
+        self.termination_event = termination_event
+        if not self.termination_event:
+            self.termination_event = custer_tester.db_cluster.nemesis_termination_event
+
+    def start(self):
+        if self.commitlog_params.use_hard_size_limit:
+            self.log.debug("starting CommitLogCheckThread")
+            self._thread.start()
+        else:
+            self.log.debug(
+                "CommitLogCheckThread was not started due to commitlog_use_hard_size_limit is %s",
+                self.commitlog_params.use_hard_size_limit)
+
+    def join(self, timeout=None):
+        return self._thread.join(timeout)
+
+    def run(self):
+        self.log.debug("CommitLogCheckThread Started")
+        try:
+            thread_end_time = time.time() + self.test_duration
+
+            self.start_time = time.time()
+
+            while time.time() < thread_end_time and not self.termination_event.is_set():
+                interval_end_time = self.start_time + self.check_interval
+
+                while time.time() < interval_end_time and not self.termination_event.is_set():
+                    self.termination_event.wait(60)
```

Comment:
can you explain this, it's not clear what this loop is doing...

Reply:
if I read it correctly it works every minute ? isn't that a bit too excessive ?

Reply:
the whole play with saving clocktime, and playing with it's clear

why not just wait the interval ?

Reply:
this was put to be able exit from thread every 60s(when test finished and termination_event is set) not every 600s 

but according to [doc](https://docs.python.org/3/library/threading.html#event-objects) it should work also as 
 self.termination_event.wait(self.check_interval) will test it

Reply:
you can see all the other places we have a temnication_event and a loop, like in Nemesis thread, that we do exactly that

Reply:
updated